core/src/acc_text/nlg/utils/nlp.clj

```diff
@@ -12,6 +12,14 @@
 
 (defn token-type [token] (if (word? token) "WORD" "PUNCTUATION"))
 
+(defn capitalize-first-word [[head & tail]]
+  (str/join [(str/capitalize head) (apply str tail)]))
+
+(defn wrap-sentence [s]
+  (str (str/trim s) "."))
+
 (defn process-sentence [s]
-  (reduce str (when-not (str/blank? s)
-                (str/join [(str/capitalize (first s)) (apply str (rest s)) \.]))))
+  (if-not (str/blank? s)
```

Comment:
why not 
```
(if (str/blank? s)
""
(the-other-thing))
```
``

Reply:
Just to keep main logic on first branch, and corner case on `else`